mod_amity/amity.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import print_function
+import random
+
+from mod_amity.models import Office, LivingSpace, Fellow, Staff, Role
+
+
+class Amity(object):
+    """
+    Amity is the main class
+    inherited classes: Room and subclass
+                        Person and subclass
+    from models.py. It also creates and perform operations to manage allocations on the available rooms in amity
+    """
+
+    def __init__(self):
+        self.living_spaces = dict(available=[], unavailable=[])
+
+        self.offices = dict(available=[], unavailable=[])
+        self.fellows = []
+        self.staff = []
+        self.ids = dict(fellow=[0], staff=[0])
+
+        self.allocated_staff = []
+        self.allocated_fellows = []
+
+    def create_office(self, name):
+        if self.get_rooms(name) is not None:
+            raise ValueError("Room with same name exists")
+        self.offices["available"].append(Office(name))
+
+    def create_living_space(self, name):
+        if self.get_rooms(name) is not None:
+            raise ValueError("Room with same name exists")
+        self.living_spaces["available"].append(LivingSpace(name))
+
+    def create_fellow(self, name, accommodation='N'):
+        fellow = Fellow(name, accommodation=accommodation, id=self.generate_fellow_id())
+        self.fellows.append(fellow)
+        self.allocate_person(fellow)
+
+        return fellow
+
+    def create_staff(self, name):
+        staff = Staff(name, id=self.get_staff_id())
+        self.staff.append(staff)
+        self.allocate_person(staff)
+
+        return staff
+
+    def allocate_person(self, person):
+        """
+        assign space to fellow/staff as requested from the available rooms, as follows
+            Staff: Office
+            Fellow: Office, Living Space(if requested)
+        :param person: created person instance
+        :return: person object with allocations, if any
+        """
+        office = random.choice(self.offices["available"]) if len(self.offices["available"]) > 0 else None
+
+        if office is not None:
+            office.allocate_space(person)
+            person.assign_office(office.name)
+
+        if person.role == Role.FELLOW:
+            living_space = random.choice(self.living_spaces["available"]) \
+                if len(self.living_spaces["available"]) > 0 else None
+            if living_space is not None and person.accommodation == 'Y':
```

Comment:
The error message is handled by the file handling the docopts commands i.e run.py at the root folder

also `==` is used here because i need to check for equality of value